pkg/leap/leap-file.go

```diff
@@ -60,20 +63,22 @@ type LeapFile struct {
 
 func New(kubeclient *kubernetes.Clientset, namespace string) (*LeapManager, error) {
 	lm := &LeapManager{
-		UbloxLsInd: make(chan ublox.TimeLs, 2),
-		Close:      make(chan bool),
-		client:     kubeclient,
-		namespace:  namespace,
-		leapFile:   LeapFile{},
+		UbloxLsInd:   make(chan ublox.TimeLs, 2),
+		Close:        make(chan bool),
+		client:       kubeclient,
+		namespace:    namespace,
+		leapFile:     LeapFile{},
+		leapFilePath: defaultLeapFilePath,
+		leapFileName: defaultLeapFileName,
 	}
-	err := lm.PopulateLeapData()
+	err := lm.populateLeapData()
 	if err != nil {
 		return nil, err
 	}
 	return lm, nil
 }
 
-func ParseLeapFile(b []byte) (*LeapFile, error) {
+func parseLeapFile(b []byte) (*LeapFile, error) {
```

Comment:
Can you do the same validation for the prefixes that start with #?  At least validate that len(fields)>=2 so this can't segfault.

Reply:
Done